plugins/UsersManager/Controller.php

```diff
@@ -405,6 +410,24 @@ public function recordUserSettings()
         return $toReturn;
     }
 
+    public function newsletterSignup()
```

Comment:
@katebutler ideally we would put this into an API so we could implement for example the same action in the mobile app or somewhere else. (of course the mobile API would also need an API method whether it has been submitted already or not but that's not needed for now).

Once this is in the API, it'll be simply `return true/false` and it should simplify things. Only need to slightly adjust the JavaScript that calls it `like {module: 'API', method: 'UsersManager.signupNewsletter'}`